extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -160,20 +166,88 @@ export class RealTimeConfigManager {
    * @visibleForTesting
    */
   maybeExecuteRealTimeConfig(customMacros, consentState) {
-    // TODO(keithwrightbos) - allow pub to override such that some callouts are
-    // still allowed.
-    if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
-        consentState == CONSENT_POLICY_STATE.UNKNOWN) {
-      return;
-    }
     if (!this.validateRtcConfig_(this.a4aElement_.element)) {
       return;
     }
+    this.consentState_ = consentState;
+    this.modifyRtcConfigForConsentStateSettings();
     customMacros = this.assignMacros(customMacros);
     this.rtcStartTime_ = Date.now();
     this.handleRtcForCustomUrls(customMacros);
     this.handleRtcForVendorUrls(customMacros);
-    return Promise.all(this.promiseArray_);
+    return this.promiseArray_.length ? Promise.all(this.promiseArray_) :
+      null;
+  }
+
+  /**
+   * Returns whether a given callout object is valid to send an RTC request
+   * to, for the given consentState.
+   * @param {Object|string} calloutConfig
+   * @return {boolean}
+   * @visibleForTesting
+   */
+  isValidCalloutForConsentState(calloutConfig) {
+    const {sendRegardlessOfConsentState} = calloutConfig;
+    if (!isObject(calloutConfig) || !sendRegardlessOfConsentState) {
+      return false;
+    }
+
+    if (typeof sendRegardlessOfConsentState == 'boolean') {
+      return sendRegardlessOfConsentState;
+    }
+
+    if (isArray(sendRegardlessOfConsentState)) {
+      for (let i = 0; i < sendRegardlessOfConsentState.length; i++) {
+        if (this.consentState_ ==
+            CONSENT_POLICY_STATE[sendRegardlessOfConsentState[i]]) {
```

Comment:
We could warn here when a constent state given does not exist in the enum

Reply:
done

Reply:
Could do a switch here:

switch (sendRegardlessOfConsentState[i]) {
    case this.consentState_:
        return true;
    case undefined:
        Log... break;
}